plugin/web/extensions/sidebar/core/components/content/common/FieldComponent/Fields/point/PointLabel.tsx

```diff
@@ -25,7 +31,20 @@ const PointLabel: React.FC<BaseFieldProps<"pointLabel">> = ({ value, editMode, o
 
   const handleFieldChange = useCallback(
     (e: ChangeEvent<HTMLInputElement>) => {
-      updatePointLabelByProp("field", e.target.value);
+      const fieldValue = e.target.value;
+
+      if (fieldValue.startsWith("=")) {
+        let extractedValue = fieldValue;
+        if (fieldValue.startsWith("={") && fieldValue.endsWith("}")) {
+          extractedValue = fieldValue.substring(2, fieldValue.length - 1);
+        } else if (fieldValue.startsWith("=${") && fieldValue.endsWith("}")) {
+          extractedValue = fieldValue.substring(3, fieldValue.length - 1);
+        } else {
+          extractedValue = fieldValue.substring(1);
+        }
+        setConditions(generateConditions(extractedValue));
+      } else setConditions(undefined);
```

Comment:
Similar to the `generateConditions` function, can you pull out the stripping of the field value in to a function of its own and return extractedValue.
Then the code here would be something like
```
const extractedValue = sanitize(fieldValue)
const conditions = generateConditions(extractedValue)
setConditions(conditions)
```
Also, put `generateConditions` and the new helper function (doesn't have to be named `sanitize`, there is probably a better name for it :P) outside of the react component. Doesn't need to be rerendered!